script/os-autoinst-setup-multi-machine

```diff
@@ -17,18 +17,59 @@ configure_firewall() {
     systemctl enable --now firewalld
     firewall-cmd --permanent --get-services | grep -q isotovideo && firewall-cmd --permanent --delete-service=isotovideo
     firewall-cmd --permanent --new-service isotovideo
-    for i in $(seq 1 $instances); do firewall-cmd --permanent --service=isotovideo --add-port=$((i * 10 + 20003))/tcp ; done
+    for i in $(seq 1 "$instances"); do firewall-cmd --permanent --service=isotovideo --add-port=$((i * 10 + 20003))/tcp ; done
     firewall-cmd --permanent --zone="$zone" --add-service=isotovideo
     firewall-cmd --get-default-zone | grep -q "$zone" || firewall-cmd --set-default-zone="$zone"
     systemctl reload firewalld
 }
 
-setup_multi_machine() {
+setup_multi_machine_with_networkmanager() {
+    # Restart NM to load ovs plugin
+    systemctl restart NetworkManager
     systemctl enable --now openvswitch
-    ovs-vsctl list-br | grep -q $bridge && ovs-vsctl del-br $bridge
-    ovs-vsctl add-br $bridge
+    # Delete any previsous connections
+    nmcli con | grep -oP 'ovs-(interface|port|bridge|slave)-[\w-]+' | xargs nmcli con del
+    # Create bridge, port and interface connection
+    nmcli con add type ovs-bridge con.int "$bridge"
+    nmcli con add type ovs-port con.int "$bridge" con.master "$bridge"
+    nmcli con add type ovs-interface con.int "$bridge" con.master "$bridge" ipv4.method manual ipv4.address 10.0.2.2/15 con.zone "$zone"
     echo "OS_AUTOINST_USE_BRIDGE=$bridge" > /etc/sysconfig/os-autoinst-openvswitch
-    cat > /etc/sysconfig/network/ifcfg-$bridge <<EOF
+    # Create tap interfaces
+    for i in 0 $(seq 1 "$instances"; seq 64 $((64+instances)); seq 128 $((128+instances))); do
+        nmcli con add type ovs-port con.int "tap$i" con.master "$bridge"
+        nmcli con add type tun mode tap owner "$(id -u _openqa-worker)" group "$(getent group nobody | cut -f3 -d:)" con.int "tap$i" master "tap$i"
+    done
+    cat > /etc/firewalld/zones/"$zone".xml <<EOF
+<?xml version="1.0" encoding="utf-8"?>
+<zone target="ACCEPT">
+  <short>"${zone^}"</short>
+  <description>All network connections are accepted.</description>
+  <service name="isotovideo"/>
+  <interface name="$bridge"/>
+  <interface name="ovs-system"/>
+  <interface name="$ethernet"/>
+  <masquerade/>
+</zone>
+EOF
```

Comment:
It would be nice to avoid duplicating the firewalld part as it looks identical in both cases.